src/Data/Formatter/DateTime.purs

```diff
@@ -173,7 +230,7 @@ formatterFParser cb =
     , (PC.try $ PS.string "S") *> map MillisecondsShort cb
     , (Placeholder <$> placeholderContent <*> cb)
     , (PS.eof $> End)
-    ]
+    ] <|> (P.fail "Format contains invalid string")
```

Comment:
`<?>` prepends `"Expected "` and didn't thought about good message which was started with `Expected` but "Expected to contain only valid characters"  might be fine here.